app/process/watcher/prometheus/watcher.go

```diff
@@ -73,37 +103,256 @@ func (pw Watcher) Watch(q qi.Queue) {
+				wrappedInstance, e := wtoken.NewWtoken(common.HexToAddress(asset), evm)
+				if e != nil {
+					panic(e)
+				}
+				name, e := wrappedInstance.Name(&bind.CallOpts{})
+				if e != nil {
+					panic(e)
+				}
+				address := pw.configuration.Bridge.EVMs[chainId].RouterContractAddress
+
+				pw.registerEvmAssetSupplyMetric(asset, name)
+				pw.registerEvmAssetBalanceMetric(asset, name, address)
+				pw.registerEvmAssetCountMetric(asset, name, address)
+			}
+		}
+	}
+}
+
+func (pw Watcher) registerHederaAssetsSupplyMetrics(asset string, res *model.TokenResponse) {
+	name := fmt.Sprintf("%s%s",
+		constants.SupplyAssetMetricNamePrefix,
+		pw.tokenIDtoMetricName(asset))
+	help := fmt.Sprintf("%s%s",
+		constants.SupplyAssetMetricsHelpPrefix,
+		res.Name)
+	pw.initAndRegAssetMetric(asset, pw.supplyAssetsMetrics, name, help)
+}
+
+func (pw Watcher) registerHbarSuppyMetric(asset string) {
+	name := fmt.Sprintf("%s%s",
+		constants.SupplyAssetMetricNamePrefix,
+		pw.tokenIDtoMetricName(asset))
+	help := fmt.Sprintf("%s%s",
+		constants.SupplyAssetMetricsHelpPrefix,
+		asset)
+	pw.initAndRegAssetMetric(asset, pw.supplyAssetsMetrics, name, help)
+}
+
+func (pw Watcher) registerBridgeAccAssetsMetrics(asset string) {
+	name := fmt.Sprintf("%s%s",
+		constants.BridgeAccAssetMetricsNamePrefix,
+		pw.tokenIDtoMetricName(asset))
+	help := fmt.Sprintf("%s%s",
+		constants.BridgeAccAssetMetricsNameHelp,
+		asset)
+	pw.initAndRegAssetMetric(asset, pw.bridgeAccAssetsMetrics, name, help)
+}
+
+func (pw Watcher) registerEvmAssetSupplyMetric(asset string, name string) {
+	assetMetricName := fmt.Sprintf("%s%s",
+		constants.SupplyAssetMetricNamePrefix,
+		pw.tokenIDtoMetricName(asset))
+	help := fmt.Sprintf("%s%s", constants.SupplyAssetMetricsHelpPrefix, name)
+	pw.initAndRegAssetMetric(asset, pw.supplyAssetsMetrics, assetMetricName, help)
+}
+
+func (pw Watcher) registerEvmAssetBalanceMetric(asset string, name string, address string) {
+	metricName := fmt.Sprintf("%s%s",
+		constants.BalanceAssetMetricNamePrefix,
+		pw.tokenIDtoMetricName(asset))
+	help := fmt.Sprintf("%s%s%s%s",
+		constants.BalanceAssetMetricHelpPrefix,
+		name,
+		constants.AssetMetricHelpSuffix,
+		address)
+	pw.initAndRegAssetMetric(asset, pw.balanceAssetsMetrics, metricName, help)
+}
+
+func (pw Watcher) registerEvmAssetCountMetric(asset string, name string, address string) {
+	metricName := fmt.Sprintf("%s%s",
+		constants.CountAssetMetricNamePrefix,
+		pw.tokenIDtoMetricName(asset))
+	help := fmt.Sprintf("%s%s%s%s",
+		constants.CountAssetMetricHelpPrefix,
+		name,
+		constants.AssetMetricHelpSuffix,
+		address)
+	pw.initAndRegAssetMetric(asset, pw.countAssetsMetrics, metricName, help)
+}
+
+func (pw Watcher) tokenIDtoMetricName(id string) string {
+	replace := strings.Replace(id, constants.DotSymbol, constants.ReplaceDotSymbol, constants.DotSymbolRep)
+	result := fmt.Sprintf("%s%s", constants.AssetMetricsNamePrefix, replace)
+	return result
+}
```

Comment:
This does not require to be associated to the struct

Reply:
Updated